lib/Backend/DecompileFunction.cpp

```diff
@@ -1200,52 +1163,43 @@ CCodeGenerator::getInstructionToken(const llvm::Instruction *I) const {
       } else {
         // If we're not doing eq or neq, we have to make sure that the
         // signedness is compatible, otherwise it would break semantics.
-        using model::PrimitiveKind::Signed;
-        using model::PrimitiveKind::Unsigned;
-        auto ICmpKind = ICmp->isSigned() ? Signed : Unsigned;
-
-        auto TargetType = model::QualifiedType(Model.getPrimitiveType(ICmpKind,
-                                                                      ByteSize),
-                                               {});
-        if (OpType0.isPointer()) {
-          Op0Token = buildCastExpr(Op0Token, OpType0, TargetType);
-        } else {
-          const model::TypeDefinition *TheType = peelConstAndTypedefs(OpType0)
-                                                   .UnqualifiedType()
-                                                   .getConst();
-          revng_assert(isa<model::PrimitiveDefinition>(TheType)
-                       or isa<model::EnumDefinition>(TheType));
-          const auto *Primitive = dyn_cast<model::PrimitiveDefinition>(TheType);
-          if (nullptr == Primitive) {
-            const auto *Enum = cast<model::EnumDefinition>(TheType);
-            const auto
-              *Underlying = Enum->UnderlyingType().UnqualifiedType().getConst();
-            Primitive = cast<model::PrimitiveDefinition>(Underlying);
+        auto TargetType = ICmp->isSigned() ?
+                            model::PrimitiveType::makeSigned(ByteSize) :
+                            model::PrimitiveType::makeUnsigned(ByteSize);
+
+        auto ToToken = [&, this](const model::Type &Type, std::string &&Token) {
```

Comment:
NOTETOSELF: review side-by-side

Reply:
@pfez can you review this?

Reply:
I opened another thread. The code you commented upon is obsolete and has been changed by Djordje recently with the work on implicit C casts. I'll review it after this PR has been rebased on current `develop`. In its current status reviewing this specific part is a waste of time.